### DISCUSSION TITLE
More advanced dropdown

### DISCUSSION THREAD

**Discussion by Dimonka2** [alpinejs/alpine#866] [1. Help] [answered]
The dropdown example in documentation works well with some limitations. Here is the code to remind how it looks like:

```html
<div x-data="dropdown()">
    <button x-on:click="open">Open</button>

    <div x-show="isOpen()" x-on:click.away="close">
        // Dropdown
    </div>
</div>
```
My problem is how to implement a dropdown, that may contain a form inside the `// Dropdown` part with another dropdown? In this case `x-on:click.away` should be smart and do not close this dropdown if the "away" click is happening inside the form in `// Dropdown` content.  

Does anyone solved this issue or have some ideas how to fix it? 
Thanks!

**Comment by KevinBatdorf** [comment, marked as answer]
Do you have an example of it not working? That should work since you are still clicking inside the top level component.

Demo: https://codepen.io/KevinBatdorf/pen/gOMdxxK

**Reply by Dimonka2** [reply]
You are right, it was because I had `@click="close()"` inside a dropdown part. Now I made differentiation between dropdown with items and dropdown with a form.
